include/dali/imgcodec/decode_results.h

```diff
@@ -86,12 +126,27 @@ class DLL_PUBLIC FutureDecodeResults {
     return *this;
   }
 
+  /**
+   * @brief Waits for all results to be ready
+   */
   void wait_all() const;
 
+  /**
+   * @brief Waits for any results that have appeared since the previous call to wait_new
+   *        (or any results, if this is the first call).
+   *
```

Comment:
I think it might be nice to have that remark that empty result here indicates that we processed everything.